codestyle/checkstyle.xml

```diff
@@ -325,5 +325,11 @@ codestyle/checkstyle.xml.&#10;"/>
 
     <module name="StringLiteralEquality"/>
 
+    <module name="Regexp">
+      <property name="format" value="Collections\.(EMPTY_LIST|EMPTY_MAP|EMPTY_SET)"/>
+      <property name="message" value="Use Collections.empty*() method instead of Collection.EMPTY_* field"/>
+      <property name="illegalPattern" value="true"/>
+    </module>
```

Comment:
I don't think this rule lines up with the intelliJ inspection warning. 

See https://github.com/JetBrains/intellij-community/blob/master/plugins/InspectionGadgets/test/com/siyeh/igtest/migration/collections_field_access_replaceable_by_method_call/CollectionsFieldAccessReplaceableByMethodCall.java

It appears that object comparisons against `Collections.EMPTY_*` are not reported as errors on purpose, but this change would make checkstyle fail.

The intelliJ inspections run as part of the build, so I think we can omit this change to the checkstyle.

Reply:
In this case, you are right. Looks like it's not easy to make identical to IntelliJ inspection check by regexp.

But generally, additional checkstyle rule makes sense sometimes, since not all developers use IntelliJ products and checkstyle allows to verify code locally, before submitting to CI. 

I've reverted checkstyle rule.

Reply:
yeah I like the checkstyle rule too, because I almost never run intelliJ inspections locally. But since they added an explicit test that verifies that style is allowed, it made me think they must have a reason to do so 🤷‍♂️